stdlib/public/core/Sort.swift.gyb

```diff
@@ -72,58 +72,105 @@ func _insertionSort<C>(
   }
 }
 
+/// Sorts the elements at indices `a`, `b`, and `c`. Stable.
+///
+/// - Precondition: `a < b && b < c`
+/// - Postcondition: `elements[a] <= elements[b] && elements[b] <= elements[c]`
+public // @testable
+func _sort3<C>(
+  _ elements: inout C,
+  _ a: C.Index, _ b: C.Index, _ c: C.Index
+  ${", by areInIncreasingOrder: (C.Iterator.Element, C.Iterator.Element) -> Bool" if p else ""}
+)
+  where
+  C : MutableCollection & RandomAccessCollection
+  ${"" if p else ", C.Iterator.Element : Comparable"}
+{
+  // Possible starting states for elements at a, b, and c:
+  // 123, 132, 213, 231, 312, 321, 112, 121, 211, 122, 212, 221, 111
```

Comment:
I've added some more comments here and above—pls let me know if that's what you were looking for.